firebase_admin/_auth_utils.py

```diff
@@ -323,3 +340,72 @@ def _build_error_message(code, exc_type, custom_message):
         exc_type and hasattr(exc_type, 'default_message')) else 'Error while calling Auth service'
     ext = ' {0}'.format(custom_message) if custom_message else ''
     return '{0} ({1}).{2}'.format(default_message, code, ext)
+
+
+def emulator_host():
+    return os.environ.get('FIREBASE_AUTH_EMULATOR_HOST')
+
+
+def is_emulator_enabled():
+    return bool(emulator_host())
+
+
+class AuthResourceUrlBuilder(object):
+    """Utility class to help building resource URLs."""
+
+    def __init__(self, project_id, version='v1'):
+        """The resource URL builder constructor.
+
+        Args:
+            project_id: The resource project ID.
+            version: The endpoint API version.
+        """
+        self.project_id = project_id
+        self.version = version
+        self._url_format = (
+            FIREBASE_AUTH_EMULATOR_BASE_URL_FORMAT.format(host=emulator_host())
+            if is_emulator_enabled() else FIREBASE_AUTH_BASE_URL_FORMAT)
+
+    def get_url(self, api='', **params):
+        """Returns the resource URL corresponding to the provided parameters.
+
+        Args:
+            api: The backend API name (optional).
+            **params: The optional additional parameters to substitute in the URL path.
+
+        Returns:
+            string: The corresponding resource URL.
+        """
+        return self._url_format.format(
+            api=api, project_id=self.project_id, version=self.version, **params)
+
+
+class TenantAwareAuthResourceUrlBuilder(AuthResourceUrlBuilder):
+    """Tenant aware resource builder utility."""
+
+    def __init__(self, project_id, version, tenant_id):
+        """The tenant aware resource URL builder constructor.
+
+        Args:
+            project_id: The resource project ID.
+            version: The endpoint API version.
+            tenant_id: The tenant ID.
+        """
+        super(TenantAwareAuthResourceUrlBuilder, self).__init__(project_id, version)
+        self.tenant_id = tenant_id
+        self._url_format = (
+            FIREBASE_AUTH_EMULATOR_TENANT_URL_FORMAT.format(host=emulator_host())
+            if is_emulator_enabled() else FIREBASE_AUTH_TENANT_URL_FORMAT)
+
+    def get_url(self, api='', **params):
+        """Returns the resource URL corresponding to the provided parameters.
+
+        Args:
+            api: The backend API name (optional).
+            **params: The optional additional parameters to substitute in the URL path.
+
+        Returns:
+            string: The corresponding resource URL.
+        """
+        return super(TenantAwareAuthResourceUrlBuilder, self).get_url(
```

Comment:
You can probably just substitute the tenant_id in the constructor. Then you don't need to override this method.

Reply:
Yup, this is addressed with the new "way" I've decided to handle the URL building.